com.unity.shadergraph/Editor/Data/Util/StackPool.cs

```diff
@@ -1,22 +1,18 @@
 using System.Collections.Generic;
+using UnityEngine.Pool;
 
 namespace UnityEditor.Graphing
 {
     static class StackPool<T>
```

Comment:
This could be simplified a lot by being defined like the built-in pools in `UnityEngine.Pool`

Reply:
There's no "trivial" conversion of the StackPool container to something based on UnityEngine.Pool.CollectionPool that I can see.  Also, as there is only one use of StackPool, I wonder if a refactor of GraphDataUtils to use a different container might make more sense.

Reply:
(eventually, but maybe not part of this PR)

Reply:
Sounds good! :)